frontend/src/App.js

```diff
@@ -1,22 +1,51 @@
-import fetchAllRoutes from './utils/fetchAllRoutes';
 import "./App.css";
 import Dropdown from "./components/Dropdown";
-import { useEffect, useState } from 'react';
+import fetchAllRoutes from "./utils/fetchAllRoutes";
+import { useEffect, useState } from "react";
 
 function App() {
   const [allRoutes, setAllRoutes] = useState([]);
- 
+  const [selectedRoute, setSelectedRoute] = useState('');
+  const [routeDetails, setRouteDetails] = useState(null);
+
   useEffect(() => {
-    const fetchRoutes = async () => {
-      const routes = await fetchAllRoutes();
-      setAllRoutes(routes);
+    const fetch = async () => {
+      try {
+        const fetchedRoutes = await fetchAllRoutes();
+        setAllRoutes(fetchedRoutes);
+      } catch (error) {
+        console.error("Failed to fetch routes:", error);
+      }
     };
-    fetchRoutes();
+    fetch();
   }, []);
 
+  useEffect(() => {
+    if (selectedRoute) {
+      const route = allRoutes.find(route => route.routeId === parseInt(selectedRoute, 10));
+      setRouteDetails(route);
+    }
+  }, [selectedRoute, allRoutes]);
```

Comment:
Mate, this is really interesting stuff - it's really cool how you got to this 👍 

But you are missing a chance to make it simpler.

You've set a new state that you don't actually need. So let's try to refactor to make it easier for us to read and maintain this code later on!

We don't need this useEffect if you implement the other changes I'm leaving in the comments